### PR TITLE
[SYCLomatic]Enable migration for CUBLASLT_MATMUL_DESC_BIAS_DATA_TYPE & BIAS_POINTER

### PR DESCRIPTION
**Pull request by abhilash1910**
@zhiweij1 @zhimingwang36

### CODE REVIEW

File: clang/runtime/dpct-rt/include/dpct/blas_gemm_utils.hpp [hunk 1]
```diff
@@ -133,7 +134,7 @@ class matmul_desc_t {
   };
 
   matmul_desc_t(compute_type compute_type, library_data_t scale_type)
-      : _compute_type(compute_type), _scale_type(scale_type) {}
+      : _compute_type(compute_type), _scale_type(scale_type), _bias_type(bias_type) {}
```
Comment (zhiweij1) [inline clang/runtime/dpct-rt/include/dpct/blas_gemm_utils.hpp]: There is no `bias_type` in the constructor parameter, so you can give a default value directly:
```suggestion
      : _compute_type(compute_type), _scale_type(scale_type), _bias_type(some default value) {}
```

Comment (abhilash1910) [reply clang/runtime/dpct-rt/include/dpct/blas_gemm_utils.hpp]: Ok Wiill modify this

Comment (abhilash1910) [reply clang/runtime/dpct-rt/include/dpct/blas_gemm_utils.hpp]: I changed to library_data_t  for bias_type (similar to scalar type).
@zhiweij1  could you please suggest ? Thanks

Comment (zhiweij1) [reply clang/runtime/dpct-rt/include/dpct/blas_gemm_utils.hpp]: 71bac7a is right. But you have not correct the init list of the constructor. Another option is removing `, _bias_type(bias_type)` in the init list and set the default value at line 179.

**Review by zhiweij1 — CHANGES_REQUESTED** [review]
Please update the PR.

File: clang/runtime/dpct-rt/include/dpct/blas_gemm_utils.hpp (hunk 1 above)
Comment (abhilash1910) [reply clang/runtime/dpct-rt/include/dpct/blas_gemm_utils.hpp]: Yes fixed, thanks

**Review by zhiweij1 — APPROVED** [review]

File: clang/runtime/dpct-rt/include/dpct/blas_gemm_utils.hpp [hunk 2]
```diff
@@ -183,6 +188,7 @@ class matmul_desc_t {
   void *_b_scale_pointer = nullptr;
   void *_d_scale_pointer = nullptr;
   void *_absmax_d_pointer = nullptr;
+  void *bias_pointer = nullptr;
```
Comment (zhiweij1) [inline clang/runtime/dpct-rt/include/dpct/blas_gemm_utils.hpp]:
```suggestion
  void *_bias_pointer = nullptr;
```

File: clang/runtime/dpct-rt/include/dpct/blas_gemm_utils.hpp [hunk 3]
```diff
@@ -174,6 +178,7 @@ class matmul_desc_t {
 
   compute_type _compute_type;
   library_data_t _scale_type;
+  library_data_t bias_type = library_data_t::real_float;
```
Comment (zhiweij1) [inline clang/runtime/dpct-rt/include/dpct/blas_gemm_utils.hpp]:
```suggestion
  library_data_t _bias_type = library_data_t::real_float;
```

**Review by zhiweij1 — CHANGES_REQUESTED** [review]
Please fix the member variable name.

**Review by zhiweij1 — APPROVED** [review]

**Review by zhimingwang36 — APPROVED** [review]

**Comment by zhimingwang36** [comment]
Please solve the conflict.

**Review by zhiweij1 — APPROVED** [review]

**Review by zhimingwang36 — APPROVED** [review]

**Pull request merged**